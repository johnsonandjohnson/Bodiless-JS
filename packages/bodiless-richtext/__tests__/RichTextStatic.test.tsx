/**
 * Copyright Â© 2019 Johnson & Johnson
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * http://www.apache.org/licenses/LICENSE-2.0
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import { mount } from 'enzyme';
import defaultValue from '../src/default-value';
import type { Value as SlateEditorValue } from '../src/Type';

const setEditMode = (isEdit: boolean) => {
  // @TODO bodiless-core internals should not be touched
  // bodiless-core should be refactored to allow injecting of default edit mode
  window.sessionStorage.isEdit = isEdit;
};
setEditMode(true);

const createRichtext = () => {
  let RichText;
  // @ts-ignore no types defined for jest.isolateModules
  jest.isolateModules(() => {
    // eslint-disable-next-line global-require
    RichText = require('../src/RichText.static').default;
  });
  return RichText as any;
};

describe('RichText', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  describe('by default', () => {
    it('produces markup that matches defined snapshot', () => {
      const RichText = createRichtext();
      const wrapper = mount(<RichText />);
      expect(wrapper.html()).toMatchSnapshot();
    });
  });
  describe('when value prop is not passed', () => {
    it('passes default value to ReactEditor', () => {
      const design = {};
      const RichText = createRichtext();
      const wrapper = mount(<RichText design={design} />);
      const editor = wrapper.find('Slate');
      const valueProp = editor.prop('value') as unknown as SlateEditorValue;
      expect(valueProp).toStrictEqual(defaultValue);
    });
  });
});
