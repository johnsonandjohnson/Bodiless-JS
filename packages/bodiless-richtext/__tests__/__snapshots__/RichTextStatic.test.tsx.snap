// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RichText by default produces markup that matches defined snapshot 1`] = `"<div role=\\"textbox\\" spellcheck=\\"false\\" autocorrect=\\"false\\" autocapitalize=\\"false\\" data-slate-editor=\\"true\\" data-slate-node=\\"value\\" contenteditable=\\"true\\" zindex=\\"-1\\" style=\\"position: relative; outline: none; white-space: pre-wrap; word-wrap: break-word; min-width: 100px;\\"><div data-slate-node=\\"element\\" style=\\"position: relative;\\"><span data-slate-node=\\"text\\"><span data-slate-leaf=\\"true\\"><span data-slate-zero-width=\\"n\\" data-slate-length=\\"0\\">ï»¿<br></span></span></span></div></div>"`;
